libs/import/src/lib/utils.ts

```diff
@@ -198,96 +199,197 @@ export function getDate(value: string, name: string): Date | ValueWithError<Date
+  const option = {
+    name: `${name} already exist`,
+    reason: `We could not create a ${name} because it already exist on the app.`,
+    message: `Please edit the corresponding sheet field with a different value.`,
+  };
+  return new ImportError(value, option);
+}
+
+export function getOptionalWarning(name: string) {
   return {
+    field: '',
+    value: undefined,
+    name: `Missing ${name}`,
+    reason: 'Optional field is missing.',
+    message: 'Fill in the corresponding sheet field to add a value.',
+    type: 'warning'
+  } as const;
+}
+
+export function optionalWarning<T = unknown>(name: string, value?: T): ImportLog<T> {
+  const option: LogOption = {
     // value is `undefined` by default because optional warning mean that the value is missing,
     // for other warning the value should passed as a parameter
-    value,
-    error: {
-      type: 'warning',
-      name: `Missing ${name}`,
-      reason: 'Optional field is missing.',
-      hint: 'Fill in the corresponding sheet field to add a value.',
-    },
+    name: `Missing ${name}`,
+    reason: 'Optional field is missing.',
+    message: 'Fill in the corresponding sheet field to add a value.',
   };
+  return new ImportWarning(value, option);
 }
 
-export function adminOnlyWarning<T = unknown>(value: T, name: string): ValueWithError<T> {
-  return {
-    value,
-    error: {
-      type: 'warning',
-      name: `${name} is only for Admins`,
-      reason: "This field is reserved for admins, it's value will be omitted.",
-      hint: 'Remove the corresponding sheet field to silence this warning.',
-    },
+export function adminOnlyWarning<T = unknown>(value: T, name: string): ImportLog<T> {
+  const option: LogOption = {
+    reason: "This field is reserved for admins, it's value will be omitted.",
+    message: 'Remove the corresponding sheet field to silence this warning.',
+    name: 'Admin only warning'
+  };
+  return new ImportWarning(value, option);
+}
+
+export function wrongValueWarning<T = unknown>(value: T, name: string, wrongData: string[]): ImportLog<T> {
+  const option: LogOption = {
+    name: `Wrong ${name}`,
+    reason: `Be careful, ${wrongData.length} values were wrong and will be omitted.`,
+    message: `${wrongData.slice(0, 3).join(', ')}...`
   };
+  return new ImportWarning(value, option);
+}
+
+export interface LogOption {
+  name: string;
+  reason: string;
+  message: string;
+  field?: string;
+}
+
+export abstract class ImportLog<T> extends Error {
+  reason: string;
+  field?: string;
+  message: string;
+  abstract type: 'warning' | 'error';
+
+  constructor(private value: T, options: LogOption) {
+    super(options.message);
+    const { message, name, reason, field = '' } = options;
+    this.name = name;
+    this.reason = reason;
+    this.field = field;
+    this.message = message;
+  }
+}
+
+export class ImportError<T> extends ImportLog<T> {
+  public type = 'error' as const;
```

Comment:
Prefer readonly over `as const`
```suggestion
  public readonly type = 'error';
```